Common/Source/Calc/Pirker.cpp

```diff
@@ -8,16 +8,105 @@
 
 #include "externs.h"
 #include "McReady.h"
+#include "CalcTask.h"
+#include "AATDistance.h"
 
-//
-// Herbert Pirker calculation
-//
-double PirkerAnalysis(NMEA_INFO* Basic, DERIVED_INFO* Calculated,
-                      const double this_bearing, const double GlideSlope) {
-  //  bool maxfound = false;
-  //  bool first = true;
+extern double CRUISE_EFFICIENCY;
+
+double ComputeTaskRequiredAltitude(NMEA_INFO* Basic, DERIVED_INFO* Calculated,
+                                   double this_maccready) {
+  double LegToGo = 0;
+  double LegBearing = 0;
+
+  const GeoPoint cur_pos = GetCurrentPosition(*Basic);
+  const GeoPoint next_pos = GetTurnpointTarget(ActiveTaskPoint);
+  cur_pos.Reverse(next_pos, LegBearing, LegToGo);
+
+  if (ActiveTaskPoint < 1) {
+    if (ValidTaskPoint(ActiveTaskPoint + 1)) {  // BUGFIX 091221
+      LegToGo = 0;
+    }
+  }
+
+  double TaskAltitudeRequired = 0;
+
+  // Calculate Final Glide To Finish
+  int FinalWayPoint = getFinalWaypoint();
+
+  double final_height =
+      FAIFinishHeight(Basic, Calculated, -1);
+  double total_energy_height =
+      Calculated->NavAltitude + Calculated->EnergyHeight;
+  double height_above_finish = total_energy_height - final_height;
+
+  TaskAltitudeRequired = final_height;
+
+  // Now add it for remaining waypoints
+  int task_index = FinalWayPoint;
+
+  while ((task_index > ActiveTaskPoint) && (ValidTaskPoint(task_index))) {
+    bool this_is_final = (task_index == FinalWayPoint) || ForceFinalGlide;
+
+    GeoPoint p0 = GetTurnpointTarget(task_index - 1);
+    GeoPoint p1 = GetTurnpointTarget(task_index);
+
+    double NextLegDistance, NextLegBearing;
+    p0.Reverse(p1, NextLegBearing, NextLegDistance);
+
+    double this_LegTimeToGo;
+    double LegAltitude = GlidePolar::MacCreadyAltitude(
+        this_maccready, NextLegDistance, NextLegBearing, Calculated->WindSpeed,
+        Calculated->WindBearing, 0, 0, this_is_final, &this_LegTimeToGo,
+        height_above_finish, CRUISE_EFFICIENCY);
+
+    TaskAltitudeRequired += LegAltitude;
+
+    if (true || ISPARAGLIDER) {
```

Comment:
always true, need to be removed ( or reason to not remove must be esplained in comment )